plugins/common/opcua/input/input_client.go

```diff
@@ -541,15 +543,52 @@ func (o *OpcUAInputClient) UpdateNodeValue(nodeIdx int, d *ua.DataValue) {
 
 func (o *OpcUAInputClient) MetricForNode(nodeIdx int) telegraf.Metric {
 	nmm := &o.NodeMetricMapping[nodeIdx]
-	fields := make(map[string]interface{})
 	tags := map[string]string{
 		"id": nmm.idStr,
 	}
 	for k, v := range nmm.MetricTags {
 		tags[k] = v
 	}
 
-	fields[nmm.Tag.FieldName] = o.LastReceivedData[nodeIdx].Value
+	fields := make(map[string]interface{})
+	if o.LastReceivedData[nodeIdx].Value != nil {
+		// Simple scalar types can be stored directly under the field name while
+		// arrays (see 5.2.5) and structures (see 5.2.6) must be unpacked.
+		// Note: Structures and arrays of structures are currently not supported.
+		if o.LastReceivedData[nodeIdx].IsArray {
+			switch typedValue := o.LastReceivedData[nodeIdx].Value.(type) {
+			case []uint8:
+				fields = unpack(nmm.Tag.FieldName, typedValue)
+			case []uint16:
+				fields = unpack(nmm.Tag.FieldName, typedValue)
+			case []uint32:
+				fields = unpack(nmm.Tag.FieldName, typedValue)
+			case []uint64:
+				fields = unpack(nmm.Tag.FieldName, typedValue)
+			case []int8:
+				fields = unpack(nmm.Tag.FieldName, typedValue)
+			case []int16:
+				fields = unpack(nmm.Tag.FieldName, typedValue)
+			case []int32:
+				fields = unpack(nmm.Tag.FieldName, typedValue)
+			case []int64:
+				fields = unpack(nmm.Tag.FieldName, typedValue)
+			case []float32:
+				fields = unpack(nmm.Tag.FieldName, typedValue)
+			case []float64:
+				fields = unpack(nmm.Tag.FieldName, typedValue)
+			case []string:
```

Comment:
One more question here. Is there a chance we see a boolean array, i.e. `[]bool` here?